internals-js/src/specs/sourceSpec.ts

```diff
@@ -172,17 +175,35 @@ export class SourceSpecDefinition extends FeatureDefinition {
             }
           });
         }
+        if (featureUrl && featureUrl.name === 'federation') {
+          federationVersion = featureUrl.version;
+        }
       });
 
+    if (result.sourceAPI || result.sourceType || result.sourceField) {
+      // Since this subgraph uses at least one of the @source{API,Type,Field}
+      // directives, it must also use v2.7 or later of federation.
+      if (!federationVersion || federationVersion.lt(this.minimumFederationVersion)) {
```

Comment:
I don't think this error is necessary. I think it's fine to just not validate rather than throw an error. Not sure that this could ever happen.

Reply:
All directives have an issue where if you don't specify the right version composition will say that it doesn't understand the directive. Not sure why this is a special case.

Reply:
because the source directives are imported from their own link instead of from the federation link. it's a little weird, but i'm ok with leaning into it because enforcing a minimum version of 2.7 should minimize the number of unhappy paths:

|  | `@link` federation < 2.7 | `@link` federation >= 2.7 |
| -- | ------ | ------ |
| composition < 2.7 | *using source is silently ignored* (🦶🏻 🔫 , but can't fix) | composition fails due to link being ahead of composition |
| composition >= 2.7 | **this scenario** which we're disallowing | happy path |

Reply:
Ok, I got it. Can you make an issue to solve this for the general case? Might be a good starter project for our new hire.